ospf6d/ospf6_message.h

```diff
@@ -22,6 +22,8 @@
 #define OSPF6_MESSAGE_H
 
 #define OSPF6_MESSAGE_BUFSIZ  4096
+#define IPPROTO_OSPF 89
```

Comment:
This define looks like it may be provided by headers in `/usr/include` on some systems, please add an `#ifndef IPPROTO_OSPF` around it.  (Also, why do we have it now but didn't need it before?)

Reply:
As per the RFC 7166, there is a requirement were u calculate checksum when auth trailer is not present and when its present u need to ignore. so u have to do checksum calculation based on config. so had to add the checksum calculation code. its required for that.

Reply:
Okay, but we already have:
```
lib/libospf.h:#ifndef IPPROTO_OSPFIGP
lib/libospf.h:#define IPPROTO_OSPFIGP         89
lib/libospf.h:#endif /* IPPROTO_OSPFIGP */
```

Please use that.

Reply:
Done